src/onboarding/onboarding-restore-backup.ts

```diff
@@ -1,136 +1,328 @@
+          : nothing
+      }
+      ${
+        ["details", "restore"].includes(this._view) && this._backup
+          ? html`<div class="backup-summary-wrapper">
+              <ha-backup-details-summary
+                translation-key-panel="page-onboarding.restore"
+                show-upload-another
+                .backup=${this._backup}
+                .localize=${this.localize}
+                @show-backup-upload=${this._reupload}
+              ></ha-backup-details-summary>
+            </div>`
+          : nothing
+      }
     `;
   }
 
-  private _back(): void {
-    navigate(`${location.pathname}?${removeSearchParam("page")}`);
-  }
+  protected firstUpdated(changedProps) {
+    super.firstUpdated(changedProps);
 
-  private _backupUploaded(ev) {
-    const backup = ev.detail.backup;
-    this._backupSlug = backup.slug;
-    this._showBackupDialog();
+    this._loadBackupInfo();
   }
 
-  private _backupCleared() {
-    this._backupSlug = undefined;
-  }
+  private async _loadBackupInfo() {
+    const backupId = localStorage.getItem(STORAGE_BACKUP_ID_KEY);
+    const restoreRunning = localStorage.getItem(STORAGE_RESTORE_RUNNING);
 
-  protected firstUpdated(changedProps) {
-    super.firstUpdated(changedProps);
-  }
+    try {
+      const {
+        last_non_idle_event: lastNonIdleEvent,
+        state: currentState,
+        backups,
+      } = await fetchBackupOnboardingInfo();
+
+      this._backupInfo = {
+        state: currentState,
+        last_non_idle_event: lastNonIdleEvent,
+      };
 
-  private async _checkRestoreStatus(): Promise<void> {
-    if (this._restoring) {
-      try {
-        await fetchInstallationType();
-      } catch (err: any) {
+      if (backupId) {
+        this._backup = backups.find(({ backup_id }) => backup_id === backupId);
+      }
+
+      const failedRestore =
+        lastNonIdleEvent?.manager_state === "restore_backup" &&
+        lastNonIdleEvent?.state === "failed";
+
+      if (failedRestore) {
+        this._failed = true;
+      }
+
+      if (restoreRunning) {
+        this._view = "status";
+        if (failedRestore || currentState !== "restore_backup") {
+          this._failed = true;
+          localStorage.removeItem(STORAGE_RESTORE_RUNNING);
+        } else {
+          this._scheduleLoadBackupInfo();
+        }
+        return;
+      }
+
+      if (!lastNonIdleEvent || !backupId) {
+        this._view = "upload";
+        return;
+      }
+
+      if (
+        this._backup &&
+        (lastNonIdleEvent.manager_state === "receive_backup" || failedRestore)
+      ) {
+        if (!this.supervisor && this._backup.homeassistant_included) {
+          this._selectedData = {
+            homeassistant_included: true,
+            folders: [],
+            addons: [],
+            homeassistant_version: this._backup.homeassistant_version,
+            database_included: this._backup.database_included,
+          };
+          this._view = "restore";
```

Comment:
Why do we go to the `restore` view when it failed? Shouldnt it go to `status`?

Reply:
This handles failedRestore when the restore request wasn't successful, e.g. when a encryption key is wrong. But I'll refactor this section and add comments.